code/modules/events/apc_short.dm

```diff
@@ -63,67 +64,54 @@
-				if(A.operating)
-					A.toggle_breaker()
+			// if it was operating, toggle off the breaker
+			if(A.operating)
+				A.toggle_breaker()
 			// no matter what, ensure the area knows something happened to the power
 			current_area.power_change()
 			affected_apc_count++
-	log_and_message_admins("APC Short Out event shorted out [affected_apc_count] APCs.")
+	log_and_message_admins("APC Short Out event has shorted out [affected_apc_count] APCs.")
 
-/proc/power_restore(announce=TRUE)
-	switch(alert("What Would You Like to Do?", "Make All Areas Powered", "Power all APCs", "Repair all APCs", "Repair and Power APCs"))
-		if("Power all APCs")	//Power without Repairing
-			if(announce)
-				GLOB.event_announcement.Announce("All operational APCs on [station_name()] have been fully charged.", "Power Systems Nominal", new_sound = 'sound/AI/poweron.ogg')
-			var/affected_apc_count = 0
-			for(var/obj/machinery/power/apc/A in GLOB.apcs)
-				var/area/current_area = get_area(A)
-				if(!is_station_level(A.z))
-					continue
-				if((length(A.wires.cut_wires) == 0) && A.operating && !A.shorted)
-					var/obj/item/stock_parts/cell/C = A.get_cell()
-					if(C)
-						C.charge = C.maxcharge
-				affected_apc_count++
-				current_area.power_change()
-			log_and_message_admins("Power has been restored to [affected_apc_count] APCs.")
-		if("Repair all APCs")	//Repair without charging
-			if(announce)
-				GLOB.event_announcement.Announce("All APCs on [station_name()] have been repaired.", "Power Systems Nominal", new_sound = 'sound/AI/poweron.ogg')
-			for(var/thing in GLOB.apcs)
-				var/obj/machinery/power/apc/A = thing
-				var/area/current_area = get_area(A)
-				if(!is_station_level(A.z))
-					continue
-				if(A.wires)
-					A.wires.repair()
-				if(!A.operating)
-					A.toggle_breaker()
-				if(A.shorted)
-					A.shorted = FALSE
-				current_area.power_change()
-			log_and_message_admins("Power has been restored to all APCs.")
-		if("Repair and Power APCs")	//Repair and Power APCs
-			if(announce)
-				GLOB.event_announcement.Announce("All APCs on [station_name()] have been repaired and recharged. We apologize for the inconvenience.", "Power Systems Optimal", new_sound = 'sound/AI/poweron.ogg')
-			// repair the APCs and recharge them
-			for(var/thing in GLOB.apcs)
-				var/obj/machinery/power/apc/A = thing
-				var/area/current_area = get_area(A)
-				if(!is_station_level(A.z))
-					continue
-				if(A.wires)
-					A.wires.repair()
-				if(!A.operating)
-					A.toggle_breaker()
-				if(A.shorted)
-					A.shorted = FALSE
-				var/obj/item/stock_parts/cell/C = A.get_cell()
-				if(C)
-					C.charge = C.maxcharge
-				current_area.power_change()
-			log_and_message_admins("Power has been restored to all APCs.")
+/proc/power_restore(announce=TRUE, var/power_type)
+	var/type = power_type //Using if() statements instead of a Switch since ifs are supposed to be slightly faster and it's for three case types
+	if(type == 0)	//Power without Repairing
```

Comment:
I'd say ditch the `type` var and just use `power_type` directly.

Reply:
Oh sure!